example/feature_demo/marshal_test.go

```diff
@@ -0,0 +1,64 @@
+package example
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/infobloxopen/protoc-gen-gorm/types"
+)
+
+func TestUnmarshalTypes(t *testing.T) {
+	ts, _ := ptypes.TimestampProto(time.Date(2009, 11, 17, 20, 34, 58, 651387237, time.UTC))
+	marshaller := &runtime.JSONPb{OrigName: true, EmitDefaults: true}
+	for in, expected := range map[string]TestTypes{
+		`{}`: {},
+		`{"api_only_string":"important text"}`:            {ApiOnlyString: "important text"},
+		`{"numbers":[1,2,3,4]}`:                           {Numbers: []int32{1, 2, 3, 4}},
+		`{"optional_string":"something real"}`:            {OptionalString: &wrappers.StringValue{Value: "something real"}},
+		`{"becomes_int":"GOOD"}`:                          {BecomesInt: 1},
+		`{"uuid":"123abc-unchecked-uuid"}`:                {Uuid: &types.UUIDValue{Value: "123abc-unchecked-uuid"}},
```

Comment:
looks like you're trimming whitespace -- you should add tests with leading and trailing whitespace (if trimming it is even a good idea, which I'm not sure)

Reply:
Are you referring to the strings.Trim? Because as above that's for the double-quotes. But I should add some tests for things like that, I agree.

Reply:
ah, turns out I wasn't paying enough attention again. In this case, I'm assuming it'd also remove quotes from within the value, as well? `"some"really_bad_uuid_value"` -> `somereally_bad_uuid_value`

Reply:
https://golang.org/pkg/strings/#Trim Just leading and trailing quotes. In theory though `"""super quoted string""""""` -> `super quoted string`

Reply:
Should we trim whitespace? Whitespace isn't a valid part of UUIDs, is it?

Reply:
my thought is we leave the input untouched -- so if they somehow put whitespace in it, we should preserve that. I didn't read your code properly (my fault) and thought that was what you were doing, but in fact you're just removing the quotes. Since we add the quotes in the first place, it makes sense to remove them too.

This is still a lazy way to do it because of the possibility of removing more than the leading and trailing quote. This is only an issue if we in fact don't validate the UUID during the un/marshal process.